redbot/core/core_commands.py

```diff
@@ -515,30 +515,31 @@ def maybe_reload(new_name):
     async def _set(self, ctx):
         """Changes Red's settings"""
         if ctx.invoked_subcommand is None:
-            admin_role_id = await ctx.bot.db.guild(ctx.guild).admin_role()
-            admin_role = discord.utils.get(ctx.guild.roles, id=admin_role_id)
-            mod_role_id = await ctx.bot.db.guild(ctx.guild).mod_role()
-            mod_role = discord.utils.get(ctx.guild.roles, id=mod_role_id)
-            prefixes = await ctx.bot.db.guild(ctx.guild).prefix()
-            if not prefixes:
-                prefixes = await ctx.bot.db.prefix()
-            locale = await ctx.bot.db.locale()
-
-            settings = (
-                "{} Settings:\n\n"
-                "Prefixes: {}\n"
-                "Admin role: {}\n"
-                "Mod role: {}\n"
-                "Locale: {}"
-                "".format(
-                    ctx.bot.user.name,
-                    " ".join(prefixes),
-                    admin_role.name if admin_role else "Not set",
-                    mod_role.name if mod_role else "Not set",
-                    locale,
+            if ctx.guild:
+                admin_role_id = await ctx.bot.db.guild(ctx.guild).admin_role()
+                admin_role = discord.utils.get(ctx.guild.roles, id=admin_role_id)
+                mod_role_id = await ctx.bot.db.guild(ctx.guild).mod_role()
+                mod_role = discord.utils.get(ctx.guild.roles, id=mod_role_id)
+                prefixes = await ctx.bot.db.guild(ctx.guild).prefix()
+                if not prefixes:
+                    prefixes = await ctx.bot.db.prefix()
```

Comment:
Perhaps we should be using `bot.get_prefix(ctx.message)` instead of reading directly from Config? I know it's not exactly the scope of this PR but I'm happy to have it done here as a minuscule change.

Reply:
Probably not. bot.get_prefix(ctx.message) will include both mention forms if the bot is mentionable, neither of which will display properly in the box formatting, and including both may be seen as confusing to users. I will be updating this with the changes allowing the global information to still be displayed.